packages/patterns/src/components/create-pattern-modal.js

```diff
@@ -42,6 +42,44 @@ export default function CreatePatternModal( {
 	const { saveEntityRecord, invalidateResolution } = useDispatch( coreStore );
 	const { createErrorNotice } = useDispatch( noticesStore );
 
+	const { corePatternCategories, userPatternCategories } = useSelect(
+		( select ) => {
+			const { getUserPatternCategories, getBlockPatternCategories } =
+				select( coreStore );
+
+			return {
+				corePatternCategories: getBlockPatternCategories(),
+				userPatternCategories: getUserPatternCategories(),
+			};
+		}
+	);
+
+	const categoryMap = useMemo( () => {
```

Comment:
One question with merging is do we dedupe by label or slug and do we prioritise the core label/slug combinations over the user ones, or the other way around? 

I think we should dedupe by label as below as that is what the user sees, so avoids confusion as currently, the likes of the core `Headers/header` label slug combination could lead to a user adding a `Headers/headers` combination and two `Headers` labels appearing. But, I don't have a strong opinion on whether the users categories should have priority over core or not.

Reply:
I don't have a preference regarding deduping the labels or the slugs.

I don't think the micro-optimization matters here for performance until proven otherwise either. 👍

Reply:
> I think we should dedupe by label as below as that is what the user sees

This logic makes sense to me 👍